src/dispatch/task/flows.py

```diff
@@ -100,76 +97,40 @@ def send_task_notification(
 
 def create_or_update_task(db_session, incident, task: dict, notify: bool = False):
     """Creates a new task in the database or updates an existing one."""
-    incident_task = task_service.get_by_resource_id(db_session=db_session, resource_id=task["id"])
-
-    assignees = []
-    for a in task["assignees"]:
-        assignees.append(
-            db_session.merge(
-                incident_add_or_reactivate_participant_flow(
-                    a, incident_id=incident.id, db_session=db_session
-                )
-            )
-        )
-
-    description = task["description"][0]
-    status = TaskStatus.open if not task["status"] else TaskStatus.resolved
-    resource_id = task["id"]
-    weblink = task["web_link"]
-
-    # TODO we can build this out as our scraping gets more advanced
-    tickets = [
-        ticket_service.get_or_create_by_weblink(db_session=db_session, weblink=t["web_link"])
-        for t in task["tickets"]
-    ]
-
-    if incident_task:
-        # allways update tickets and assignees
-        incident_task.assignees = assignees
-        incident_task.tickets = tickets
+    existing_task = task_service.get_by_resource_id(
+        db_session=db_session, resource_id=task["resource_id"]
+    )
 
-        # only notify if it's newly resolved
-        if status == TaskStatus.resolved:
-            if incident_task.status != TaskStatus.resolved:
-                incident_task.status = status
+    if existing_task:
+        existing_status = existing_task.status  # fetch old task data
+        task = task_service.update(
+            db_session=db_session, task=existing_task, task_in=TaskUpdate(**task)
+        )
 
-                if notify:
+        if notify:
+            # determine if task was perviously resolved
+            if existing_status == TaskStatus.resolved:
+                # determine if we have a newly resolved task (since last sync)
+                if existing_task.status != TaskStatus.resolved:
```

Comment:
I still don't think this is going to work as intended. If a task goes from open (`existing_status == TaskStatus.open`) to resolved (`existing_task.status == TaskStatus.resolved`), you're not notifying. You're notifying when a task goes from resolved to open. Previously, we [checked](https://github.com/Netflix/dispatch/blob/develop/src/dispatch/task/flows.py#L132) if the new status was resolved and then if the previous one was open (or not resolved).

Reply:
I've inverted the statements.